File: app/src/main/java/com/adiaz/deportesmadrid/adapters/GenericAdapter.java
package com.adiaz.deportesmadrid.adapters;

import android.content.Context;
import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import com.adiaz.deportesmadrid.R;
import com.adiaz.deportesmadrid.utils.ListItem;

import java.util.List;

import butterknife.BindView;
import butterknife.ButterKnife;

public class GenericAdapter extends RecyclerView.Adapter<GenericAdapter.ViewHolder> {

    Context mContext;
    List<ListItem> sportsList;
    private final ListItemClickListener mOnClickListener;

    public GenericAdapter(Context mContext, ListItemClickListener listItemClickListener, List<ListItem> sportsList) {
        this.mContext = mContext;
        this.mOnClickListener = listItemClickListener;
        this.sportsList = sportsList;
    }

    @Override
    public ViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
        LayoutInflater layoutInflater = LayoutInflater.from(mContext);
        View view = layoutInflater.inflate(R.layout.listitem_competition, parent, false);
        return new ViewHolder(view);
    }

    @Override
    public void onBindViewHolder(ViewHolder holder, int position) {
        String str = sportsList.get(position).getName() + " - " + sportsList.get(position).getCount();
        holder.tvCompetitionTitle.setText(str);
        //holder.tvCompetitionCount.setText(sportCount);
    }

    @Override
    public int getItemCount() {
        return sportsList.size();
    }

    public interface ListItemClickListener {
        void onListItemClick(int clickedItemIndex);
    }

    public class ViewHolder extends RecyclerView.ViewHolder implements View.OnClickListener {

        @BindView(R.id.tv_sport_title)
        TextView tvCompetitionTitle;

        /*        @BindView(R.id.tv_sport_count)
        TextView tvCompetitionCount;*/

        public ViewHolder(View itemView) {
            super(itemView);
            ButterKnife.bind(this, itemView);
            itemView.setOnClickListener(this);
        }

        @Override
        public void onClick(View view) {
            int position = getAdapterPosition();
            mOnClickListener.onListItemClick(position);
        }
    }
}

File: app/src/main/java/com/adiaz/ligasmadrid/adapters/GenericAdapter.kt
package com.adiaz.ligasmadrid.adapters

import android.content.Context
import android.support.v7.widget.CardView
import android.support.v7.widget.RecyclerView
import android.view.View
import android.view.ViewGroup
import android.widget.TextView

import com.adiaz.ligasmadrid.R
import com.adiaz.ligasmadrid.extensions.layoutInflater
import com.adiaz.ligasmadrid.utils.ListItem
import com.adiaz.ligasmadrid.utils.UtilsPreferences

import kotlinx.android.synthetic.main.listitem_generic.view.*

class GenericAdapter(var mContext: Context, val mOnClickListener: ListItemClickListener, var sportsList: List<ListItem>?) :
        RecyclerView.Adapter<GenericAdapter.ViewHolder>() {

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder {
        val view = mContext.layoutInflater.inflate(R.layout.listitem_generic, parent, false)
        return ViewHolder(view)
    }

    override fun onBindViewHolder(holder: ViewHolder, position: Int) {
        var str = sportsList!![position].name
        if (UtilsPreferences.isShowCompetitionsNumber(mContext)) {
            str += " (" + sportsList!![position].count + ")"
        }
        holder.tvCompetitionTitle.text = str
    }

    override fun getItemCount(): Int {
        return sportsList!!.size
    }

    interface ListItemClickListener {
        fun onListItemClick(clickedItemIndex: Int)
    }

    inner class ViewHolder (itemView: View) : RecyclerView.ViewHolder(itemView), View.OnClickListener {

        var tvCompetitionTitle: TextView = itemView.listitemTitle
        var cvListItem: CardView = itemView.cardView

        init {
            cvListItem.setOnClickListener(this)
        }

        override fun onClick(view: View) {
            val position = adapterPosition
            mOnClickListener.onListItemClick(position)
        }
    }
}
